src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -296,648 +454,673 @@ def build_process_costs(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.instrumentation_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.piping_materials_and_labor_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Piping, materials and labor ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.plant_services_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Ancillary cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.electrical_materials_and_labor_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Electrical, materials and labor ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            # buildings cost variables
-            self.buildings_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Buildings cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.instrumentation_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.process_buildings_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Process buildings cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                self.plant_services_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
+                    doc="Ancillary cost in $MM",
+                    units=getattr(pyunits, "MUSD_" + CE_index_year),
+                )
 
-            self.auxiliary_buildings_costs = Var(
-                initialize=value(self.total_BEC),
-                bounds=(0, 1e4),
-                doc="Auxiliary buildings cost in $MM",
-                units=getattr(pyunits, "MUSD_" + CE_index_year),
-            )
+                # buildings cost variables
+                self.buildings_costs = Var(
+                    initialize=value(self.total_BEC),
+                    bounds=(0, 1e4),
```

Comment:
How reasonable are these bounds, especially given that the basis and units of cost are not well known?

Reply:
For millions of USD, the scale of plants should not exceed 10,000 million USD for any of the cost variables so that should be a sufficiently large upper bound. The supported USD cost years have index values between ~350 to 850, so the upper bound works for any cost year.

We would probably want to change or remove the upper bound if we're allowing cost units other than "millions of USD".

Reply:
The CEI values have been steadily increasing over the years however, so eventually they will reach a point where the bound is not reasonable (although that might not happen in a realistic timeframe). There is also the question of people wanting to use non-USD currencies which will have different values.

The real question here is not what is a reasonable value for the bounds, but whether bounds are necessary at all. I.e. given that these values are ultimately based on the sum of costs of individual pieces of equipment, and those costs are in turn based on dimensions of the equipment, these values should be implicitly bounded by the dimensions of the equipment. Adding additional bounds here is essentially adding an arbitrary bound on the dimensions of the equipment that might be tighter than necessary (or alternatively is a bound that will never be active due to tighter constraints elsewhere in the model).